cmd/osm-injector/osm-injector.go

```diff
@@ -133,13 +134,14 @@ func main() {
 		metricsstore.DefaultMetricsStore.CertIssuedTime,
 	)
 
-	// Initialize Configurator to watch osm-config ConfigMap
-	cfg := configurator.NewConfigurator(kubeClient, stop, osmNamespace, osmConfigMapName)
-	configMap, err := cfg.GetConfigMap()
+	// This component will be watching the OSM MeshConfig and will make it
+	// to the rest of the components.
+	cfg := configurator.NewConfiguratorWithCRDClient(configClientset.NewForConfigOrDie(kubeConfig), stop, osmNamespace, osmMeshConfigName)
+	meshConfig, err := cfg.GetMeshConfigJSON()
 	if err != nil {
-		log.Error().Err(err).Msgf("Error parsing ConfigMap %s", osmConfigMapName)
+		log.Error().Err(err).Msgf("Error parsing MeshConfig %s", osmMeshConfigName)
 	}
-	log.Debug().Msgf("Initial ConfigMap %s: %s", osmConfigMapName, string(configMap))
+	log.Info().Msgf("Initial MeshConfig  %s: %v", osmMeshConfigName, meshConfig)
```

Comment:
We should not be dumping the entire MeshConfig resource. This could have user specific configuration values that might be sensitive. In the past we agreed to not do so.